src/driver.rs

```diff
@@ -948,4 +1049,17 @@ impl ProcessingSession {
         self.io.mem.files.borrow_mut().remove(&self.tex_xdv_path);
         Ok(0)
     }
+
+
+    /// Consume this session and return the current set of files in memory.
+    ///
+    /// This convenience function tries to help with the annoyances of getting
+    /// access to the in-memory file data after the engine has been run.
+    pub fn into_file_data(self) -> HashMap<OsString, Vec<u8>> {
+        // There must be a better way to do this ... Note that you *cannot*
```

Comment:
This avoids the temporary. It can panic though:
```rust
Rc::try_unwrap(self.io.mem)
  .expect("multiple strong refs to MemoryIo files")
  .into_inner()
```

Reply:
I think I like this more ... avoids rebuilding the hashmap. And I am pretty sure that it will never panic unless the caller has made their own copy of `files`, in which case they're getting what they deserve.